pom.xml

```diff
@@ -159,14 +190,20 @@
                 <artifactId>maven-surefire-plugin</artifactId>
                 <version>2.22.2</version>
                 <executions>
+                    <execution>
+                        <id>default-test</id>
+                        <!-- this executes the same tests as unit-tests -->
+                        <phase>none</phase>
+                    </execution>
                     <execution>
                         <id>unit-tests</id>
                         <phase>test</phase>
                         <configuration>
-
                             <excludes>
                                 <exclude>**/*IT.java</exclude>
                             </excludes>
+                            <!-- use JaCoCo agent configuration specific to unit tests -->
+                            <argLine>${jacoco-agent.unit-tests}</argLine>
```

Comment:
💬 Pass the JaCoCo agent configuration for unit tests to the unit tests execution. This ensures that this goal execution will write coverage data to a dedicated file and it won't overlap with the coverage data for the integration-tests execution.